src/pages/[marketplace]/deals/index.tsx

```diff
@@ -140,11 +140,27 @@ const Deals: NextPageWithLayout = () => {
 		[marketplace, publicKey]
 	);
 
+	const hasActiveCredixPass = (credixPass: CredixPass) => credixPass && credixPass.isActive;
+	const isBorrowerAndNotUnderWriter = (credixPass: CredixPass) =>
+		credixPass?.isBorrower && !credixPass.isUnderwriter;
+
+	const isAllowedToSeeDeal = (deal: Deal) => {
+		if (!hasActiveCredixPass(credixPass)) {
+			return false;
+		}
+
+		if (isBorrowerAndNotUnderWriter(credixPass)) {
```

Comment:
why is this an exclusive or?

Reply:
switch the checks because it's a bit confusing imo.
first check if the user is an underwriter and return true if so, then check if the deal belongs to the user or not